src/bot/config.ts

```diff
@@ -0,0 +1,31 @@
+import yargs = require('yargs');
+import fs = require('fs');
+
+/**
+ * Gets the specified key from a JSON, raising an exception if the value is null/undefined
+ */
+function getNonNull(jsonObject: any, key: string) {
+    if (jsonObject[key] == null) {
+        throw new Error('Missing required configuration key: ' + key);
+    }
+    return jsonObject[key];
+}
+
+class Configuration {
+    token: string;
+    owner: string;
+    commandPrefix: string;
+    statusType: string;
+    statusText: string;
+
+    constructor(jsonObject: any) {
+        this.token = getNonNull(jsonObject, 'token');
+        this.owner = getNonNull(jsonObject, 'owner');
+        this.commandPrefix = getNonNull(jsonObject, 'commandPrefix');
+        this.statusType = getNonNull(jsonObject, 'statusType');
+        this.statusText = getNonNull(jsonObject, 'statusText');
+    }
+}
+
+const rawConfig = fs.readFileSync(yargs.argv.config as string, 'utf8');
+export default new Configuration(JSON.parse(rawConfig));
```

Comment:
I think the best spot for this would be `src/lib/classes/`.

Reply:
I'm not 100% on what the dividing line is for the lib folder vs bot. Should I just put the Configuration class in `src/lib/classes` and leave the initialization here?

Reply:
That's a good point to bring up.  I've been bouncing back and forth on structure.

There are classes that are under bot folder, which could go under the lib directory as well.  I think we should have a quick discord conversation before moving forward much further about our structure.